app/src/main/java/net/gaast/giggity/Schedule.java

```
@@ -421,6 +424,155 @@ private void loadIcal(BufferedReader in) {
+			JSONArray events = conference.getJSONArray("sessions");
+
+			for (i = 0; i < events.length(); i++) {
+
+				JSONObject event = events.getJSONObject(i);
+				String uid = event.getString("id");
+				String title = event.getString("title");
+
+                /*Our date format is different and I changed getTimeInMillis() a bit to ignore "+08"
+                in second part to avoid error in integer parsing*/
+				String startTimeS = event.getString("start_time");
+				String endTimeS = event.getString("end_time");
+				Date startTime, endTime;
+				startTime = new Date(getTimeInMillis(startTimeS));
+				endTime = new Date(getTimeInMillis(endTimeS));
+				Schedule.Item item = new Schedule.Item(uid, title, startTime, endTime);
+				item.setDescription(event.getString("long_abstract"));
+
+				if (event.getString("signup_url") != "null") {
+					item.addLink(new Link(event.getString("signup_url")));
+				}
+
+				JSONObject microlocation = event.getJSONObject("microlocation");
+				String location = microlocation.getString("name");
+
+				if ((line = tentMap.get(location)) == null) {
+					line = new Schedule.Line(location, location);
+					tents.add(line);
+					tentMap.put(location, line);
+				}
+
+				//Getting value (latitude and longitude) from the map by key (name)
+				String locString = locs.get(line.getTitle());
+				String latitude = locString.substring(0, locString.indexOf(','));
+				String longitude = locString.substring(locString.indexOf(',') + 1);
+
+				//Adding location details here
+				String latlon = null;
+				try {
+					latlon = ("geo:0,0?q=" + longitude + "," +
+							latitude + "(" +
+							URLEncoder.encode(line.getTitle(), "utf-8") + ")");
+				} catch (UnsupportedEncodingException e) {
+					e.printStackTrace();
+				}
+
+
+				if (latlon != null) {
+					line.setLocation(latlon);
+				}
+
+				line.addItem(item);
+			}
+
+		} catch (JSONException e) {
+			e.printStackTrace();
```

Should report a load error to the user as well, similar to how the other loaders do it.